### PR TITLE
#1252 - Fix timeline highlights

### LINKED ISSUE #1252: Highlights are broken for an empty profile on the Timeline page
This is with weighted highlights turned on.

### PR DESCRIPTION
**Pull request by piatra**
Fixes #1252

This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/1266)

### CODE REVIEW

File: content-src/selectors/selectors.js [hunk 1]
```diff
@@ -77,7 +87,7 @@ module.exports.selectNewTabSites = createSelector(
     // don't dedupe against stuff that isn't shown
     let [topSitesRows, spotlightRows] = dedupe.group([TopSites.rows.slice(0, TOP_SITES_LENGTH), Spotlight.rows]);
 
-    // Find the index of the recommendation. If we have an index, find that recommmendation and put it
+    // Find the index of the recommendation. If we have an index, find that recommendation and put it
```
Comment (sarracini) [inline content-src/selectors/selectors.js]: recommendation isn't spelled with 3 m's?!?!? 😏

File: content-src/selectors/selectors.js [hunk 2]
```diff
@@ -15,18 +15,28 @@ function isValidSpotlightSite(site) {
     site.title !== site.description);
 }
 
+const selectWeightedHighlights = module.exports.selectWeightedHighlights = createSelector(
+  [
+    state => state.WeightedHighlights
+  ],
+  WeightedHighlights => {
+    const rows = WeightedHighlights.rows.length > 5 ? WeightedHighlights.rows : WeightedHighlights.rows.concat(firstRunData.Highlights);
```
Comment (sarracini) [inline content-src/selectors/selectors.js]: why 5?

Comment (piatra) [inline content-src/selectors/selectors.js]: as discussed over vidyo. will fix.

**Comment by piatra** [comment]
@sarracini Updated, please review.

**Comment by sarracini** [comment]
r+ 👍

**Pull request merged**